### PR TITLE
Adopt Kotlin/JS for 1.4-M2

### PR DESCRIPTION
**Pull request by ilgonmic**
- Migrate names to 1.4-M2 in both mode with compatibility with 1.3.7x versions

### CODE REVIEW

File: gradle/compile-js-multiplatform.gradle [hunk 1]
```diff
@@ -5,15 +5,14 @@
...
-            if (js.hasProperty("irTarget") && irTarget != null) {
-                irTarget.nodejs()
-                irTarget.compilations['main']?.dependencies {
-                    api "org.jetbrains.kotlinx:atomicfu-js:$atomicfu_version"
-                }
+    js {
+        if (it.hasProperty("moduleName")) {
+            moduleName = project.name
+        }
+        if (it.hasProperty("irTarget") && it.irTarget != null) {
```
Comment (qwwdfsad) [inline gradle/compile-js-multiplatform.gradle:12]: when writing non-trivial construct please drop a comment why it is needed

File: gradle/compile-js-multiplatform.gradle [hunk 2]
```diff
@@ -5,15 +5,14 @@
...
-            // Enable built-in test runner only for IR target.
-            // These runners don't support changing js module name change.
-            if (js.hasProperty("irTarget") && irTarget != null) {
-                irTarget.nodejs()
-                irTarget.compilations['main']?.dependencies {
-                    api "org.jetbrains.kotlinx:atomicfu-js:$atomicfu_version"
-                }
+    js {
+        if (it.hasProperty("moduleName")) {
+            moduleName = project.name
```
Comment (qwwdfsad) [inline gradle/compile-js-multiplatform.gradle:10]: and here as well please

File: gradle/compile-js-multiplatform.gradle (hunk 1 above)
Comment (ilgonmic) [reply gradle/compile-js-multiplatform.gradle:20]: Done

File: gradle/compile-js-multiplatform.gradle (hunk 2 above)
Comment (ilgonmic) [reply gradle/compile-js-multiplatform.gradle:12]: Done

File: gradle/compile-js-multiplatform.gradle (hunk 1 above)
Comment (ilgonmic) [reply gradle/compile-js-multiplatform.gradle:20]: Done

File: gradle/compile-js-multiplatform.gradle (hunk 2 above)
Comment (ilgonmic) [reply gradle/compile-js-multiplatform.gradle:12]: Done

**Pull request merged**